controllers/dscinitialization/utils.go

```
@@ -217,120 +270,121 @@ func (r *DSCInitializationReconciler) reconcileDefaultNetworkPolicy(ctx context.
+								MatchLabels: map[string]string{
+									"kubernetes.io/metadata.name": "openshift-host-network",
 								},
 							},
 						},
 					},
-					{
-						From: []networkingv1.NetworkPolicyPeer{
-							{ // need this for cluster-monitoring work: cluster-monitoring->ODH namespaces
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{
-										"kubernetes.io/metadata.name": "openshift-monitoring",
-									},
+				},
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{ // need this for cluster-monitoring work: cluster-monitoring->ODH namespaces
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"kubernetes.io/metadata.name": "openshift-monitoring",
 								},
 							},
 						},
 					},
 				},
-				PolicyTypes: []networkingv1.PolicyType{
-					networkingv1.PolicyTypeIngress,
-				},
 			},
-		}
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+			},
+		},
+	}
 
-		// Create NetworkPolicy if it doesn't exist
-		foundNetworkPolicy := &networkingv1.NetworkPolicy{}
-		justCreated := false
-		err := r.Client.Get(ctx, client.ObjectKeyFromObject(desiredNetworkPolicy), foundNetworkPolicy)
-		if err != nil {
-			if k8serr.IsNotFound(err) {
-				// Set Controller reference
-				err = ctrl.SetControllerReference(dscInit, desiredNetworkPolicy, r.Scheme)
-				if err != nil {
-					log.Error(err, "Unable to add OwnerReference to the Network policy")
-					return err
-				}
-				err = r.Client.Create(ctx, desiredNetworkPolicy)
-				if err != nil && !k8serr.IsAlreadyExists(err) {
-					return err
-				}
-				justCreated = true
-			} else {
+	// Create NetworkPolicy if it doesn't exist
+	foundNetworkPolicy := &networkingv1.NetworkPolicy{}
+	justCreated := false
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(desiredNetworkPolicy), foundNetworkPolicy)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
```

this can probably be simplified by failing fast  if `!k8serr.IsNotFound(err)`